src/components/generics/generic-toggle-button.tsx

```diff
@@ -0,0 +1,48 @@
+import { Variant } from "@mui/material/styles/createTypography";
+import { ComponentType, Dispatch, ReactNode, SetStateAction } from "react";
+import { ButtonIconProps } from "../../types";
+import { Button, Typography } from "@mui/material";
+
+/**
+ * Component properties
+ */
+interface ToggleButtonProps {
+  children?: ReactNode;
+  buttonVariant?: "text" | "contained" | "outlined";
+  titleVariant?: Variant;
+  value: boolean;
+  setValue: Dispatch<SetStateAction<boolean>>;
+  ButtonIcon?: ComponentType<ButtonIconProps>;
+  title?: string;
+}
+/**
+ * FormOpen Toggle Button component
+ *
+ * @param props CreateNewButtonProps
+ */
+const GenericToggleButton = (props: ToggleButtonProps) => {
```

Comment:
This seems like a leftover name. Did you mean to use the 'FormOpen' name in the comment? I'd love to have a configurable and themable GenericToggleButton though

Reply:
Fixed the comment. Also this component could be expanded on. It's missing a feature to change the button style onClick or on value change. This feature could be optional and activated/deactivated with a boolean prop.